tests/unit/s2n_locking_test.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include <pthread.h>
+
+#include "crypto/s2n_locking.h"
+
+#define LOCK_N 1
+
+#if !(S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0))
+
+static void* s2n_test_thread(void *arg)
```

Comment:
Honestly I have trouble understanding what's happening in this test thread function. I guess we're attempting to change the value of some passed flag? My small nit is that maybe this should be called passed_flag, b/c the word "lock" is losing all meaning when I read this test.

Reply:
Haha I actually mean the flag as in "code execution has passed the lock / advanced past the lock", I didn't consider that we were also "pass"ing the flag the function. I'll see if I can use a clearer name.